src/Queue/Jobs/RabbitMQJob.php

```diff
@@ -108,15 +107,9 @@ public function delete(): void
      */
     public function release($delay = 0): void
     {
-        parent::release($delay);
+        parent::release();
 
-        if ($delay > 0) {
```

Comment:
This was an intended behaviour in order attempts to work as expected.

Reply:
Because my solution always pushes a 'new' message to the queue by calling the laterRaw(). When the delay is zero or in the past, the 'new' message is pushed with pushRaw() 

[See](https://github.com/adm-bome/laravel-queue-rabbitmq/blob/849d608be1ddfe81f25dbe2f49f238d38cda6054/src/Queue/RabbitMQQueue.php#L146)

In your solution a failed message was rejected/nacked and so always in front of the queue, withholding other messages to get proccesed. In my solution the message gets a higher priority, over other messages, to get processed first.

Reply:
Maybe there is an solution for loosing the messages when no DLX is set and rejecting messages, and keeping the xDeatCount property into account.

I have to test this situation.

CASE:
define queue without DLX.

When rejecting also requeue the message, by calling `$this->reject($this, true)`.

Hopefully xDeathCount is updated when nacking the message. 

The only problem i think persistst: is that when there are 1000 Jobs in the queue, the failed Job will proccess only when the other 999 are processed first. Leaving us with a situation that we can't tell when in time the message is processed again.

Reply:
Because I removed DLX on the main queue, and always publish a new message... this intention was removed.
`laravel.attempts` in the message arguments is always updated accordingly and xDeathCount was rejected.

I think this conversation can be closed.